### PR TITLE
Libmesh submodule update

### PR DESCRIPTION
**Pull request by roystgnr**
This shouldn't be merged yet, since we don't want to make @milljm 's job in #26848 even harder; plus there's another libMesh PR that I'd be happy to see snuck in under the wire if we really take too long ... but I'm anxious like to see what downstream CI makes of the changes that are here so far; it's entirely possible that we have users who are reading QuadShell4 meshes (because that's a Cubit default) and turning them second-order, and any golded results based on that will break when "full second-order" becomes *actually*-full second-order - see #27987 for the fix to such a failure in the Navier-Stokes module.

Summary of changes:

- Support for `QuadShell9` elements, and fixes for refinement of
  `QuadShell8` elements.  This is a behavior-changing update:
  `QuadShell4` elements, when elevated to second order, now become
  `QuadShell9` (with a full biquadratic Lagrange space) rather than
  `QuadShell8` (with a "Serendipity element" space).
- Fixes for libMesh `make dist` and `make distcheck` with new
  submodules and new contributed packages.
- Support and unit tests for non-square EigenSparseMatrix matrices
- Support and unit tests for reading sparse matrices from
  gzipped-Matlab format files.
- Utility for converting sparse matrix files from PETSc binary formats
  to Matlab format.
- Improved speed for reading sparse matrices.
- Refactoring of some solution transfer code; exposure of
  `MeshfreeInterpolationFunction` class to users.
- Support and unit tests in TIMPI for `set_union` of `multiset` and
  `multimap` containers, including `unordered_` versions.
- Refactored Nedelec finite element code for simplification and
  generality.
- Fixes and testing for more IsoGeometricAnalysis use cases on
  distributed meshes.
- Additional fparser headers installed, to enable user subclassing
- Assorted bug fixes: configure now supports preexisting `-Werror` in
  compiler flags (with tested compilers), more standard treatment of
  PETSc error codes.
- PetscVector subvectors can now be created by supplying only local
  subvector indices.
- Incoming vectors to `local_variable_indices()` are now cleared
  before filling.
- PerfLog profiling of matrix reads and mesh constraint
  initialization.
- Refactoring and optional debugging features for mesh constraint
  code.
- Added `RBConstruction::is_serial_training_type()` and EIM updates to
  reduced basis code.
- Added a DofMap API for users to request full matrix sparsity pattern
  retention even with linear algebra packages that do not need it
  themselves.

### CODE REVIEW

**Comment by roystgnr** [comment]
👔

**Comment by roystgnr** [comment]
Rebased and hashes redone, just to see if CI gives us any new headaches, but we'll still also want to upgrade to a slightly newer libMesh hash shortly.

File: modules/doc/content/newsletter/2024/2024_07.md [hunk 1]
```diff
@@ -9,6 +9,54 @@ for a complete description of all MOOSE changes.
...
+  initialization.
+- Refactoring and optional debugging features for mesh constraint
+  code.
+- Added `RBConstruction::is_serial_training_type()` and EIM updates to
+  reduced basis code.
+- Added a DofMap API for users to request full matrix sparsity pattern
+  retention even with linear algebra packages that do not need it
+  themselves.
+- `System::is_initialized()` is now `const`
+
```
Comment (lindsayad) [inline modules/doc/content/newsletter/2024/2024_07.md:59]:
```suggestion
- Added `LibmeshPetscCall` macro that can wrap any PETSc function call without need for `LIBMESH_CHKERR(ierr)`

```

**Review by lindsayad — COMMENTED** [review]
This will be great to get in. A lot of nice changes/additions. This is needed for the PETSc update

**Review by loganharbour — APPROVED** [review]

**Comment by loganharbour** [comment]
Feel free to merge @roystgnr if I'm not around to do it. Just check next and make sure it's passing first, if so.

**Comment by lindsayad** [comment]
Noooo I should have checked this more closely. This doesn't have libmesh/libmesh#3905

**Comment by loganharbour** [comment]
> Noooo I should have checked this more closely. This doesn't have [libMesh/libmesh#3905](https://github.com/libMesh/libmesh/pull/3905)

Put another up. Our build infrastructure for these environments has gotten a lot cleaner and I'm on board with weekly updates if there's a need.

**Comment by roystgnr** [comment]
Going to squeeze https://github.com/libMesh/libmesh/pull/3895 and hopefully an upcoming PR of my own in before the next submodule update, but that should still be this week.

**Pull request merged**